packages/manager/apps/catalog-revamp/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,205 @@
+          : selectedUniverses.includes(label || '');
+
+      const resultsNumber = count ? ` (${count})` : '';
+
+      return (
+        <div className="break-inside-avoid">
+          <OsdsCheckbox
+            checked={isChecked}
+            onOdsCheckedChange={(event: CustomEvent) =>
+              handleCheckboxChange(type, label, event)
+            }
+          >
+            <OsdsCheckboxButton
+              size={ODS_CHECKBOX_BUTTON_SIZE.sm}
+              color={ODS_THEME_COLOR_INTENT.primary}
+            >
+              <span slot="end">
+                {loading ? (
+                  <OsdsSkeleton />
+                ) : (
+                  <OsdsText
+                    color={ODS_THEME_COLOR_INTENT.default}
+                    level={ODS_TEXT_LEVEL.button}
+                    size={ODS_TEXT_SIZE._600}
+                  >
+                    {label} {resultsNumber}
+                  </OsdsText>
+                )}
+              </span>
+            </OsdsCheckboxButton>
+          </OsdsCheckbox>
+        </div>
+      );
+    };
+
+    return (
+      <>
+        <div className="filters-container filters-container grid grid-cols-1 md:flex">
+          <div className="filters-universes">
+            <OsdsText
+              level={ODS_TEXT_LEVEL.heading}
+              size={ODS_TEXT_SIZE._400}
+              color={ODS_THEME_COLOR_INTENT.text}
+              className="title"
+            >
+              {t('manager_catalog_filters_universes')}
+            </OsdsText>
+            {universes.map(
+              (item: { universe: string; count: number }, id: number) => (
+                <FilterItem
+                  key={`item-${id}`}
+                  label={item.universe}
+                  count={item.count}
+                  type="universe"
+                />
+              ),
+            )}
+            {!universes.length &&
+              [...Array(5)].map((_, id: number) => (
+                <FilterItem key={`item-${id}`} loading={true} type="universe" />
+              ))}
+          </div>
+          <div className="filters-categories grid">
+            <OsdsText
+              level={ODS_TEXT_LEVEL.heading}
+              size={ODS_TEXT_SIZE._400}
+              color={ODS_THEME_COLOR_INTENT.text}
+              className="title"
+            >
+              {t('manager_catalog_filters_categories')}
+            </OsdsText>
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3">
+              {categories.map(
+                (item: { category: string; count: number }, id: number) => (
+                  <FilterItem
+                    key={`item-${id}`}
+                    label={item.category}
+                    count={item.count}
+                    type="category"
+                  />
+                ),
+              )}
+              {!categories.length &&
+                [...Array(15)].map((_, id: number) => (
```

Comment:
why mapping on an hardcoded array of 15 ? if the API return 16 categories you could not display it

Reply:
It's 15 skeletons, there is nothing about the API response length here.